src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -409,20 +444,41 @@ void OTARequestor::OnConnectionFailure(void * context, PeerId peerId, CHIP_ERROR
     default:
         break;
     }
+
+    // Give driver a chance to schedule another query
+    requestorCore->mOtaRequestorDriver->UpdateNotFound(UpdateNotFoundReason::ConnectionFailed, chip::System::Clock::Seconds32(0));
 }
 
+// Sends the QueryImage command to the Provider currently set in the OTARequestor
+void OTARequestor::TriggerImmediateQueryInternal()
+{
+    // We are now connecting to a provider for the purpose of sending a QueryImage,
+    // treat this as a move to the Querying state
+    RecordNewUpdateState(OTAUpdateStateEnum::kQuerying, OTAChangeReasonEnum::kSuccess);
```

Comment:
We probably should be setting the state in `ConnectToProvider` based on the onConnectedAction value. And then all the sets in `OnConnected`. At the very least, we are now not transitioning states for download/apply/notify.

We could just have a follow up after your PR merges.

Reply:
Setting the states for download/apply/notify have no effect on the periodic timer because we never enter those states from kIdle, by that time the timer is stopped so from that perspective it doesn't matter when we enter those. The Querying state is special in that sense -- we must enter it as soon as possible after the driver API call because the driver stops the periodic timer right before that call thus we must immediately change from kIdle.